File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    answer= []
    gcd = 1
    for i in range(1,n+1):
        if n%i == 0 and m%i == 0:
            gcd = i
    lcm = n*m // gcd
    answer.append(gcd)
    answer.append(lcm)
    return answer

File: Java/백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();
        int x = Math.max(a, b);
        int y = Math.min(a, b);
        System.out.println(gcd(x, y));
        System.out.println(lcm(x, y));
    }

    private static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }

        return gcd(b, a%b);
    }

    private static int lcm(int a, int b) {
        if (a == 0 || b == 0) {
            return 0;
        }

        return a*b / gcd(a, b);
    }

}